flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/SinkV2ITCase.java

```diff
@@ -227,4 +353,72 @@ private void assertUnalignedCheckpointInNonSink(StreamGraph streamGraph) {
                 .allMatch(StreamEdge::supportsUnalignedCheckpoints)
                 .isNotEmpty();
     }
+
+    /**
+     * A stream source that: 1) emits a list of elements without allowing checkpoints, 2) then waits
+     * for two more checkpoints to complete, 3) then re-emits the same elements before 4) waiting
+     * for another two checkpoints and 5) exiting.
+     */
+    private Source<Integer, ?, ?> createStreamingSourceForScalingTest() {
```

Comment:
we could use this source in the other streaming tests as well (as in more recent versions of flink) but I didn't modify the other tests to minimize the diff